### ISSUE TITLE
`getChangelogEntry` should use `readChangesetState` install of reading `changelog` file?

### ISSUE THREAD

**Issue by JounQin** [changesets/action#209]
At https://github.com/stylelint/stylelint/pull/6299, we're using our own format of CHANGELOG without `major`/`minor`/`patch` sections. It results https://github.com/stylelint/stylelint/pull/6301

> This PR was opened by the [Changesets release](https://github.com/changesets/action) GitHub action. When you're ready to do a release, you can merge this and publish to npm yourself or [setup this action to publish automatically](https://github.com/changesets/action#with-publishing). If you're not ready to do a release yet, that's fine, whenever you add more changesets to main, this PR will be updated.
>
> Releases

I think `readChangesetState` can/should handle this case, because they have the meta information without generated CHANGELOG.

**Comment by bluwy** [comment]
I don't think we can address this until we can fix it in core first. We should track this in these issues:

- https://github.com/changesets/changesets/issues/959
- https://github.com/changesets/changesets/issues/1241
- https://github.com/changesets/changesets/issues/1497
